strawberry/schema/types/base_scalars.py

```diff
@@ -15,16 +15,18 @@ def inner(value: str) -> object:
         try:
             return parser(value)
         except ValueError as e:
-            raise GraphQLError(f'Value cannot represent a {type_}: "{value}". {e}')
+            raise GraphQLError(
+                f'Value cannot represent a {type_}: "{value}". {e}'
+            ) from e
 
     return inner
 
 
 def parse_decimal(value: object) -> decimal.Decimal:
     try:
         return decimal.Decimal(str(value))
-    except decimal.DecimalException:
-        raise GraphQLError(f'Value cannot represent a Decimal: "{value}".')
+    except decimal.DecimalException as e:
+        raise GraphQLError(f'Value cannot represent a Decimal: "{value}".') from e
```

Comment:
Also, I don't think GraphQL Errors should be raised from anything. The spec errors are good in their closed form, I'd prefer them to not be related to the python errors.